include/delivery.php

```diff
@@ -170,7 +171,10 @@ function delivery_run(&$argv, &$argc){
 					$item['deleted'] = 1;
 			}
 
-			if ((count($items) == 1) && ($items[0]['uri'] === $items[0]['parent-uri'])) {
+			// When commenting too fast after delivery, a post wasn't recognized as top level post.
+			// The count then showed more than one entry. The additional check should help.
+			// The check for the "count" should be superfluous, but I'm not totally sure by now, so we keep it.
+			if ((($items[0]['id'] = $item_id) OR (count($items) == 1)) AND ($items[0]['uri'] === $items[0]['parent-uri'])) {
```

Comment:
I did the change as a quick fix in the morning. I was not awake enough for understanding or even writing the other way.

BTW: I just see that I "compare" by using the "=" instead of the "==" (which is no comparism).